rig/place_and_route/place/sa/kernel.py

```diff
@@ -0,0 +1,78 @@
+"""General interface for a SA algorithm kernel."""
```

Comment:
Can't this just be in the docs? Don't we normally do that for things which would be interfaces?